build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/leafTask.ts

```diff
@@ -45,7 +45,12 @@ export abstract class LeafTask extends Task {
 	private _parentLeafTasks: Set<LeafTask> | undefined | null;
 	private parentWeight = -1;
 
-	constructor(node: BuildPackage, command: string, taskName: string | undefined) {
+	constructor(
+		node: BuildPackage,
+		command: string,
+		taskName: string | undefined,
+		private readonly isTemp: boolean = false, // indicate if the task is for temporary and not for execution.
```

Comment:
What is a temp task? How does it differ from a "group task" or "virtual task" (can't remember what you're calling them exactly) - the ones that don't execute anything themselves, they're just nodes in the graph with children tasks. How are these temp tasks different?

Reply:
I temporary create `TscTask` for project references just so I can reuse the code to check if it is up-to-date.  It is never entered into the build graph and never executed.   It probably can be refactored better, but again don't want to churn the code.

Reply:
Got it. I think the thing I don't like is that it's not localized to the tsc task (and probably can't be without more refactoring - understood).